registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java

```diff
@@ -85,604 +91,833 @@
+        Mockito.when(abisHandlerUtil.getPacketStatus(any())).thenReturn(AbisConstant.PRE_ABIS_IDENTIFICATION);
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        assertTrue(messageDto.getMessageBusAddress().toString()
+                .equalsIgnoreCase(MessageBusAddress.ABIS_HANDLER_BUS_IN.toString()));
+    }
+
+    @Test
+    public void testApisResourceAccessException() throws Exception {
+        Mockito.doThrow(new ApisResourceAccessException("Exception")).when(cbeffValidateAndVerificatonService).validateBiometrics(any(), any());
+        Mockito.when(bioDedupeService.getFileByRegId(anyString(),anyString())).thenReturn(null);
+        ApisResourceAccessException e=new ApisResourceAccessException();
+
+
+        Mockito.doThrow(e).when(utility).getApplicantAge(any(),any(),any());
+        Mockito.when(registrationStatusMapperUtil
+                .getStatusCode(RegistrationExceptionTypeCode.APIS_RESOURCE_ACCESS_EXCEPTION)).thenReturn("REPROCESS");
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+
+        assertTrue(messageDto.getIsValid());
+        assertTrue(messageDto.getInternalError());
+    }
+
+
+    /* use case 1 -> No match of RID/AID found in bio dedupe:
+        This use case is handled in noMatchedRidFound().
+
+       use case 2 -> One match of RID/AID found at bio dedupe stage. UIN of the duplicate match is same as the update packet UIN
+        This use case is handled in noMatchedRidFound() because abisHandlerUtil.getUniqueRegIds() will
+        return the empty matched rid list because it rid's associated with current packet.
+
+       use case 4 -> More than one match of RID/AID found at bio dedupe stage, all UIN of duplicate matches are same as update packet UIN:
+        This use case is handled in noMatchedRidFound() because abisHandlerUtil.getUniqueRegIds() will
+        return the empty matched rid list because it rid's associated with current packet.
+     */
+    @Test
+    public void noMatchedRidFound() throws ApisResourceAccessException, IOException,
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095011");
+        registrationStatusDto.setRegistrationType("UPDATE");
+        registrationStatusDto.setStatusCode("PROCESSING");
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        Mockito.when(abisHandlerUtil.getPacketStatus(any())).thenReturn(AbisConstant.POST_ABIS_IDENTIFICATION);
+        Set<String> matchedRidList = new HashSet<>();
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
```

Comment:
Done. It Encountered with exception while Assing the values to the Matched rid list( ArrayList<String> matchedRegIdsList = new ArrayList<String>(matchedRegIds); ) and provided the Status Comment in DTO as "Unknown exception occurred null" so asserting that only.

Reply:
If null is not possible why are checking if == null case ?

https://github.com/mosip/registration/blob/11e3ed12384c25d7c28a4e823df255ce86e70fae/registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java#L420

Reply:
Resolving it, for now because this review comment is just on the unit test case, this needs to be discussed later, @Khuddusshariff0022 can we create a task?